clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -110,7 +110,13 @@ public ConsumerCoordinator(ConsumerNetworkClient client,
 
         addMetadataListener();
 
-        this.autoCommitTask = autoCommitEnabled ? new AutoCommitTask(autoCommitIntervalMs) : null;
+        if (autoCommitEnabled) {
```

Comment:
KafkaConsumer just grabs the group ID and the auto commit flag directly from config. The default for the group ID is an empty string and the consumer is intended to work just fine in simple consumer mode without a group ID as long as you don't commit offsets. However, group IDs are required for either group membership functionality or offset commit support. Should we be validating that, if auto commit is enabled, we also have a non-empty group ID?